src/sentry/search/events/fields.py

```diff
@@ -475,7 +475,7 @@ def parse_arguments(function, columns):
     i, j = 0, 0
 
     while j < len(columns):
-        if i == j and columns[j] == '"':
+        if not quoted and columns[j] == '"':
```

Comment:
@Zylphrex do you remember why this was `i==j` originally? Tests still passed with this method so I'm worried I might be missing something

Reply:
If I'm not wrong then this `i == j` condition signals that were at the begging of the argument. If you change it to `not quoted`, then if an unescaped quote appears in the middle of the argument, it might be seen as quoting.

Reply:
do we also need to check `not escaped` here?

Reply:
This whole function breaks my brain lol. Tony do you have an example of a valid case where you can have an unescaped quote in the middle of an argument? Because wouldn't `not quoted and columns[j] == '"'` signify the beginning of an argument as well?

Basically what I see is if we encounter an unescaped quote we can do 2 things:
either set quote=True to open it and quote=False to close it out. So an unescaped character would break this in either case right?

Reply:
Also just as an aside, I had to run the code and put in a breakpoint to see what `columns` looked like- I wouldn't have guessed that it was a comma separated string from how it was named 😛

Reply:
Here's an example to illustrate the difference between the two conditions

using `i == j`
```
>>> parse_arguments("count_if", "a\",a")
['a"', 'a']
```

using `not quoted`
```
>>> parse_arguments("count_if", "a\",a")
['a",a']
```

By switching to `not quoted, it will start seeing quotes that in in the middle of the argument as an opening quote. Which in the example above, results in that comma being consumed by the first argument rather than being seen as a separator.

Reply:
from conversation in slack, wasn't able to remember specifically what the reason i had for `not quoted` but found another issue with spaces around quoted arguments. Added a new case for that in this loop

Also added a bunch of tests for parse_arguments to help in the future :+1: